acceptance/tests/loader/autoload_from_resource_type_decl.rb

```diff
@@ -4,80 +4,64 @@
     Puppet::Type.newtype(:type_tst) do
       newparam(:name, :namevar => true) do
         Puppet.notice("found_type_tst")
       end
     end
     END
-    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_1_path}", custom_type)
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_path}", custom_type)
+
+    helper_content = <<-END
+    # this is an empty file
+    END
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_helper_path}", helper_content)
 
     site_pp = <<-PP
     Resource['type_tst'] { 'found_type': }
     PP
     create_sitepp(master, tmp_environment, site_pp)
   end
 
-  on(master, "chmod -R 755 /tmp/#{tmp_environment}")
-
-  # rotate the server log so that it won't rotate during the test
-  rotate_puppet_server_log(master)
+  on(master, "chmod -R 755 '/tmp/#{tmp_environment}'")
 
+  # This will fail because the agent side will not be able to find the "required_helper.rb"
   with_puppet_running_on(master, {}) do
     agents.each do |agent|
-      on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
-         :acceptable_exit_codes => 0) do |puppet_result|
-        assert_match(/Notice: found_type_tst/, puppet_result.stdout, 'Expected to see output from new type: type_tst')
-      end
-    end
-  end
-  type_message_count = 0
-  step 'ensure that we got a message for the type' do
-    on(master, "cat #{puppet_server_log}") do |cat_result|
-      assert_match(/\[puppetserver\] Puppet found_type_tst/, cat_result.stdout, 'Expected to see entry for new type type_tst')
-      type_message_count = cat_result.stdout.split(/\[puppetserver\] Puppet found_type_tst/).length
+      on(agent, puppet("agent -t --server #{master.hostname} --environment '#{tmp_environment}'"),
+         :acceptable_exit_codes => 1)
     end
   end
 
+  # This should work because <path>/require_helper.rb is available on the server
   step 'generate pcore files' do
-    on(master, puppet("generate types --environment #{tmp_environment}")) do |puppet_result|
-      assert_match(/Notice: Generating '.*\/type_tst\.pp' using 'pcore' format/, puppet_result.stdout,
+    on(master, puppet("generate types --environment '#{tmp_environment}'")) do |puppet_result|
+      assert_match(/Notice: Generating '\/.*\/type_tst\.pp' using 'pcore' format/, puppet_result.stdout,
                    'Expected to see Generating message for type: type_tst')
       assert_match(/Notice: found_type_tst/, puppet_result.stdout, 'Expected to see generate output from new type: type_tst')
     end
   end
 
-  # restart so that we will load and use the generated types
-  on(master, "service #{master['puppetservice']} restart")
-
-  agents.each do |agent|
-    step 'rerun agents after generate, ensure proper runs' do
-      on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
+  # now that we have generated types this will work
```

Comment:
Compilation should now work since it shouldn't load the ruby type (or it's helper).